src/pages/authors.js

```diff
@@ -0,0 +1,399 @@
+        }
+    } catch (err) {
+        console.error("fetch authors error", err);
+    }
+    return authors;
+};
+
+const callSaveAuthor = async (author) => {
+    const client = createApolloClient();
+    try {
+        await client.mutate({
+            mutation: CREATE_AUTHOR,
+            variables: {
+                author: author,
+            },
+        });
+        // console.log("save author response", response);
+    } catch (err) {
+        console.error("failed to save author", err);
+    }
+    return;
+};
+
+const callUpdateAuthor = async (author) => {
+    const client = createApolloClient();
+    try {
+        await client.mutate({
+            mutation: UPDATE_AUTHOR,
+            variables: {
+                author: {
+                    id: author.id,
+                    name: author.name,
+                    biography: author.biography,
+                    born_date: author.born_date,
+                },
+            },
+        });
+        // console.log("update author response", response);
+    } catch (err) {
+        console.error("failed to update author", err);
+    }
+    return;
+};
+
+const callDeleteAuthor = async (authorid) => {
+    const client = createApolloClient();
+    try {
+        await client.mutate({
+            mutation: DELETE_AUTHOR,
+            variables: {
+                authorid: authorid,
+            },
+        });
+        // console.log("delete author response", response);
+    } catch (err) {
+        console.error("failed to delete author", err);
+    }
+    return;
+};
+
+const AuthorsComponent = (props) => {
+    const [author, setAuthor] = useState({ name: "", biography: "" });
+    const [authors, setAuthors] = useState(props.authors);
+
+    const [dialogMode, setDialogMode] = useState("create");
+
+    const [openDialog, setOpen] = useState(false);
+
+    const handleChange = (e) => {
+        const { name, value } = e.target;
+        setAuthor({ ...author, [name]: value });
+    };
+
+    const saveAuthor = async (event) => {
+        event.preventDefault();
+        await callSaveAuthor(author);
+        setAuthor({});
+        setOpen(false);
+        let updateAuthors = await fetchAuthors();
+        setAuthors(updateAuthors);
+        // console.log("save author details", author);
+    };
+
+    const updateAuthor = async (event) => {
+        event.preventDefault();
+        await callUpdateAuthor(author);
+        setAuthor({});
+        setOpen(false);
+        let updateAuthors = await fetchAuthors();
+        setAuthors(updateAuthors);
+        // console.log("update author details", author);
+    };
+
+    const openEditAuthor = (author) => {
+        setDialogMode("edit");
+        setAuthor(author);
+        setOpen(true);
+    };
+
+    const deleteAuthor = async (authorid) => {
+        await callDeleteAuthor(authorid);
+        let updateAuthors = await fetchAuthors();
+        setAuthors(updateAuthors);
+    };
```

Comment:
You can also look at `useQuery` and `useMutation` hooks. they provide a very clean interface